src/screens/discover/index.tsx

```diff
@@ -126,39 +145,125 @@ function Discover() {
 	}, [])
 
 	useEffect(() => {
-		logger.info('fetching daos')
-		fetchMoreNewDaos(true)
-		fetchMorePopularDaos(true)
-		if(scwAddress) {
-			fetchMoreMyDaos(true)
-		}
-	}, [])
+		(async() => {
+			if(isAdmin === undefined) {
+				return
+			}
 
-	useEffect(() => {
-		if(scwAddress) {
-			fetchMoreMyDaos(true)
-		}
-	}, [scwAddress])
+			await new Promise(resolve => setTimeout(resolve, 0))
+			fetchMoreDaos(true)
+			if(scwAddress) {
+				fetchMoreMyDaos(true)
+			}
+		})()
+	}, [scwAddress, isAdmin, searchString])
+
+	if(isAdmin === undefined) {
+		return (
+			<Center w='100%'>
+				<Loader />
+			</Center>
+		)
+	}
 
-	return buildComponent()
+	return (
```

Comment:
Put this in a buildComponent function in the start of the code, and return that in the end.

Reply:
Got it